### PR TITLE
Make LoadGSS more flexible

### PR DESCRIPTION
**Pull request by peterfpeterson**
IDL reduction of NOMAD data was generating different spacing in the detector position line of the GSAS files. This change broke `LoadGSS` parsing the detector position from the file.

**To test:**

Since there is a unit test added for this behavior, a code review should suffice.

_There is no associated issue_

_Does not need to be in the release notes_ because not many users will see the issue

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadGSS.cpp [hunk 1]
```diff
@@ -187,33 +194,51 @@ API::MatrixWorkspace_sptr LoadGSS::loadGSASFile(const std::string &filename,
...
       } else if (key1 == "Primary") {
         // Primary flight path ...
-        std::string s1, s2;
-        inputLine >> s1 >> s2;
-        primaryflightpath = atof(s2.c_str()); // convertToDouble(s2);
-        g_log.information() << "L1 = " << primaryflightpath << '\n';
+        boost::smatch result;
+        if (boost::regex_search(inputLine.str(), result, L1_REG_EXP) &&
+            result.size() == 2) {
+          primaryflightpath = atof(std::string(result[1]).c_str());
```
Comment (AntonPiccardoSelg) [inline Framework/DataHandling/src/LoadGSS.cpp]: Could [std::stof](http://en.cppreference.com/w/cpp/string/basic_string/stof) be used here, ie.  `primaryflightpath = stof(std::string(result[1]));`?

File: Framework/DataHandling/src/LoadGSS.cpp [hunk 2]
```diff
@@ -187,33 +194,51 @@ API::MatrixWorkspace_sptr LoadGSS::loadGSASFile(const std::string &filename,
...
+        double totalpath(0.f);
+        double tth(0.f);
+        double difc(0.f);
+
+        boost::smatch result;
+        if (boost::regex_search(inputLine.str(), result, DET_POS_REG_EXP) &&
+            result.size() == 4) {
+          totalpath = atof(std::string(result[1]).c_str());
+          tth = atof(std::string(result[2]).c_str());
+          difc = atof(std::string(result[3]).c_str());
```
Comment (AntonPiccardoSelg) [inline Framework/DataHandling/src/LoadGSS.cpp]: Could `std::stof` be used here?

**Comment by peterfpeterson** [comment]
I had no idea that `std::stof` and `std::stod` existed. I'll switch to those.

**Comment by AntonPiccardoSelg** [comment]
Changes look good to me, happy to :shipit:

**Comment by AndreiSavici** [comment]
Failed OSX build is unrelated.

**Pull request merged**